src/Particle/SoaDistanceTableAAOMPTarget.h

```diff
@@ -352,6 +359,24 @@ struct SoaDistanceTableAAOMPTarget : public DTD_BConds<T, D, SC>, public Distanc
     }
   }
 
+  void mw_updatePartial(const RefVectorWithLeader<DistanceTableData>& dt_list, IndexType jat, const std::vector<bool>& from_temp) override
+  {
+    // if temp data on host is not updated during p-by-p moves, there is no need to update distance table
+    if (!(modes_ & DTModes::NEED_TEMP_DATA_ON_HOST)) return;
+
+#pragma omp parallel for
+    for (int iw = 0; iw < dt_list.size(); iw++)
+      dt_list[iw].updatePartial(jat, from_temp[iw]);
+  }
+
+  void mw_finalizePbyP(const RefVectorWithLeader<DistanceTableData>& dt_list,
```

Comment:
I guess that means a side effect of this function is a total transfer of the distance table device to host?  Hasn't that already been done on the device? So mw_evaluate seems to got to evaluate which the base class doesn't mention doing transfer between memory spaces and neither does the implementation here. Needs docs.

Reply:
AA table is Ne^2/2 which is too much for GPU. So the full table is never stored fully on GPU. Since the "temp" data are discarded during p-by-p move, a recompute is needed to update the table on the host.